t/proto_template/.github/workflows/upload.yml

```diff
@@ -0,0 +1,70 @@
+name: 'upload'
+
+on:
+  push:
+    branches:
+      - 'main'
+  workflow_dispatch:
+
+concurrency:
+  group: '${{ github.workflow }}-${{ github.head_ref || github.ref }}'
+  cancel-in-progress: true
+
+env:
+  GO_VERSION: 'REPLACE_GO_VERSION'
+  WIF_PROVIDER: 'REPLACE_WIF_PROVIDER'
+  WIF_SERVICE_ACCOUNT: 'REPLACE_WIF_SERVICE_ACCOUNT'
+  PROJECT_ID: 'REPLACE_PROJECT_ID'
+  GO_GAR_REPOSITORY_NAME: 'REPLACE_GO_GAR_REPOSITORY_NAME'
+  GO_GAR_REPOSITORY_LOCATION: 'REPLACE_GO_GAR_REPOSITORY_LOCATION'
+  GO_MODULE_PATH: 'REPLACE_GO_MODULE_PATH'
+
+jobs:
+  ci:
+    uses: './.github/workflows/ci.yml'
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+
+  deploy_go:
+    needs:
+      - 'ci'
+    runs-on: 'ubuntu-latest'
+    permissions:
+      contents: 'read'
+      id-token: 'write'
```

Comment:
Seems these are the same permissions between jobs? You can move to workflow level.